src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -1149,7 +1163,7 @@ public override TypeWithAnnotations ParameterTypeWithAnnotations(int index)
         {
             Debug.Assert(this.HasExplicitlyTypedParameterList);
             Debug.Assert(0 <= index && index < _parameterTypesWithAnnotations.Length);
-            return _parameterTypesWithAnnotations[index];
+            return _parameterTypesWithAnnotations.IsDefault ? default : _parameterTypesWithAnnotations[index];
```

Comment:
>return _parameterTypesWithAnnotations.IsDefault ? default : _parameterTypesWithAnnotations[index]; [](start = 12, length = 98)

We are asserting that ```_parameterTypesWithAnnotations``` is not default and actually accessing its property above. I am not sure whether this change is needed.